test/test_access_om2_config.py

```diff
@@ -0,0 +1,116 @@
+from datetime import timedelta
+import re
+
+import pytest
+import f90nml
+
+from util import get_git_branch_name
+
+# Mutually exclusive topic keywords
+TOPIC_KEYWORDS = {
+    'spatial extent': {'global', 'regional'},
+    'forcing product':	{'JRA55', 'ERA5'},
+    'forcing mode': {'repeat-year', 'ryf', 'repeat-decade', 'rdf',
+                        'interannual', 'iaf'},
+    'model': {'access-om2', 'access-om2-025', 'access-om2-01'}
+}
+
+
+class AccessOM2Branch:
+    """Use the naming patterns of the branch name to infer informatiom of
+    the ACCESS-OM2 config"""
+
+    def __init__(self, branch_name):
+        self.branch_name = branch_name
+        self.set_resolution()
+
+        self.is_high_resolution = self.resolution in ['025deg', '01deg']
+        self.is_bgc = 'bgc' in branch_name
+
+    def set_resolution(self):
+        # Resolutions are ordered, so the start of the list are matched first
+        resolutions = ['025deg', '01deg', '1deg']
+        self.resolution = None
+        for res in resolutions:
+            if res in self.branch_name:
+                self.resolution = res
+                return
+        # TODO Should unknown resolutions fail the pytests or be ignored?
+
+
+@pytest.fixture(scope="class")
+def branch(control_path):
+    branch_name = get_git_branch_name(control_path)
+    assert branch_name is not None, (
+        f"Failed getting git branch name of control path: {control_path}"
+    )
+    return AccessOM2Branch(branch_name)
+
+
+@pytest.mark.access_om2
+class TestAccessOM2:
+    """ACCESS-OM2 Specific configuration and metadata tests"""
+
+    def test_mppncombine_fast_collate_exe(self, config, branch):
+        if branch.is_high_resolution:
+            pattern = r'/g/data/vk83/apps/mppnccombine-fast/.*/bin/mppnccombine-fast'
+            if 'collate' in config:
+                assert re.match(pattern, config['collate']['exe']), (
+                    "Expect collate executable set to mppnccombine-fast"
+                    )
+
+    def test_metadata_realm(self, metadata, branch):
+        expected_realms = ['ocean', 'seaIce']
+        expected_config = 'realm:\n - ocean\n - seaIce'
+        if branch.is_bgc:
+            expected_realms.append('ocnBgchem')
+            expected_config += '\n - ocnBgchem'
+
+        assert ('realm' in metadata
+                and metadata['realm'] == expected_realms), (
+                    'Expected metadata realm set to:\n' + expected_config
+                    )
```

Comment:
Is this comparison order dependent? Perhaps making them both a [`set`](https://docs.python.org/3/library/stdtypes.html#set) would be a good approach?

Reply:
Thanks, already have changed both into sets